packages/fxa-auth-server/test/local/payments/configuration/manager.js

```diff
@@ -0,0 +1,260 @@
+    termsOfService: 'https://terms.com',
+    termsOfServiceDownload: 'https://terms-download.com',
+    webIcon: 'https://web-icon.com',
+  },
+};
+
+const planConfig = {
+  active: true,
+  stripePriceId: 'mock-stripe-price-id',
+  capabilities: {
+    '*': 'stuff',
+  },
+  locales: {},
+  styles: {
+    webIconBackground: '#fff',
+  },
+  support: {},
+  uiContent: {},
+};
+
+async function deleteQueryBatch(db, query, resolve) {
+  const snapshot = await query.get();
+
+  const batchSize = snapshot.size;
+  if (batchSize === 0) {
+    // When there are no documents left, we are done
+    resolve();
+    return;
+  }
+
+  // Delete documents in a batch
+  const batch = db.batch();
+  snapshot.docs.forEach((doc) => {
+    batch.delete(doc.ref);
+  });
+  await batch.commit();
+
+  // Recurse on the next process tick, to avoid
+  // exploding the stack.
+  process.nextTick(() => {
+    deleteQueryBatch(db, query, resolve);
+  });
+}
+
+async function deleteCollection(db, collectionRef, batchSize) {
+  const query = collectionRef.orderBy('__name__').limit(batchSize);
+
+  return new Promise((resolve, reject) => {
+    deleteQueryBatch(db, query, resolve).catch(reject);
+  });
+}
+
+describe('PaymentConfigManager', () => {
+  let paymentConfigManager;
+  let testProductId;
+  let testPlanId;
+  let productConfigDbRef;
+  let planConfigDbRef;
+
+  beforeEach(async () => {
+    testProductId = randomUUID();
+    testPlanId = randomUUID();
+    const firestore = setupFirestore(mockConfig);
+    Container.set(AuthFirestore, firestore);
+    Container.set(AuthLogger, {});
+    Container.set(AppConfig, mockConfig);
+    paymentConfigManager = new PaymentConfigManager();
+    productConfigDbRef = paymentConfigManager.productConfigDbRef;
+    planConfigDbRef = paymentConfigManager.planConfigDbRef;
+    await productConfigDbRef.doc(testProductId).set(productConfig);
+    await planConfigDbRef
+      .doc(testPlanId)
+      .set({ ...planConfig, productId: 'test-product' });
+  });
+
+  afterEach(async () => {
+    const productConfigDbRef = paymentConfigManager.productConfigDbRef;
+    const planConfigDbRef = paymentConfigManager.planConfigDbRef;
+    await deleteCollection(
+      paymentConfigManager.firestore,
+      productConfigDbRef,
+      100
+    );
+    await deleteCollection(
+      paymentConfigManager.firestore,
+      planConfigDbRef,
+      100
+    );
+    Container.reset();
+  });
+
+  describe('load', async () => {
+    it('loads products and plans and returns them', async () => {
+      await paymentConfigManager.load();
+      const products = paymentConfigManager.allProducts();
+      const plans = paymentConfigManager.allPlans();
+      assert.equal(products.length, 1);
+      assert.equal(plans.length, 1);
+    });
+  });
+
+  describe('listeners', () => {
+    it('starts and stops', async () => {
+      await paymentConfigManager.startListener();
+      await paymentConfigManager.stopListener();
+    });
```

Comment:
Since there aren't any assertions in this test, what are you testing?

Reply:
That it doesn't deadlock forever, but not a great test, agreed.

Reply:
You could assert that `onSnapshot` is called twice: once for products and once for plans?
Also consider adding a test case for the error scenario where onSnapshot fails?